components/script/script_thread.rs

```diff
@@ -325,11 +325,51 @@ impl OpaqueSender<CommonScriptMsg> for Sender<MainThreadScriptMsg> {
     }
 }
 
+/// The state that a document managed by this script thread could be in.
+#[must_root]
+enum DocumentState {
+    Kept(bool, JS<Document>),
```

Comment:
Could you document what this `bool` means? I guess it's `thawed`, but would be ok to have it written here, or alternatively add a binary enum or a `Kept { thawed: bool, document: JS<Document> }` instead of a plain `bool`.